packages/core-components/src/components/EmbeddedContainer.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <ElementList
+    :add-element-options="addElementOptions"
+    :elements="embeds"
+    :enable-add="!isDisabled && enableAdd"
+    :supported-types="types"
+    @add="addItems"
+    @update="reorderItem"
+  >
+    <template #default="{ element, isDragged }">
+      <ContainedContent
+        :element="element"
+        :is-disabled="isDisabled"
+        :is-dragged="isDragged"
+        v-bind="$attrs"
+        class="my-2"
+        @delete="requestDeleteConfirmation(element)"
+        @save="save(element, 'data', $event)"
+        @save:meta="save(element, 'meta', $event)"
+      />
+    </template>
+  </ElementList>
+</template>
+
+<script lang="ts" setup>
+import { calculatePosition } from '@tailor-cms/utils';
+import cloneDeep from 'lodash/cloneDeep';
+import { computed, inject } from 'vue';
+import type { ContentElement } from '@tailor-cms/interfaces/content-element';
+import mapKeys from 'lodash/mapKeys';
+import sortBy from 'lodash/sortBy';
+
+import { useConfirmationDialog } from '../composables/useConfirmationDialog';
+import ContainedContent from './ContainedContent.vue';
+import ElementList from './ElementList.vue';
+
+interface Props {
+  types: string[];
+  container: { embeds: Record<string, ContentElement> };
+  isDisabled?: boolean;
+  enableAdd?: boolean;
+  addElementOptions?: Record<string, any>;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  isDisabled: false,
+  enableAdd: true,
+  types: () => [],
+  addElementOptions: () => ({}),
+});
+const emit = defineEmits(['save', 'delete']);
+
+const editorBus = inject<any>('$editorBus');
+const showConfirmationDialog = useConfirmationDialog();
+
+const embeds = computed(() => {
+  const items = Object.values(props.container.embeds ?? {});
+  return sortBy(items, 'position') as ContentElement[];
+});
+
+const addItems = (items: ContentElement[]) => {
+  items = Array.isArray(items) ? items : [items];
+  const container = cloneDeep(props.container);
+  container.embeds = { ...container.embeds, ...mapKeys(items, 'id') };
```

Comment:
Should we use uids (since we use that also to ref these elements in the UI)

Reply:
ids in the embeds are actually uuids https://github.com/tailor-cms/author/blob/6912c856be8287d9902cd5f71f198d3559b10568/packages/core-components/src/components/AddElement/index.vue#L189